lalrpop/src/normalize/token_check/mod.rs

```diff
@@ -21,19 +21,50 @@ mod test;
 
 pub fn validate(mut grammar: Grammar) -> NormResult<Grammar> {
     let (has_enum_token, all_literals) = {
+        let opt_match_token = grammar.match_token();
+
+        let (match_mappings, match_catch_all) = if let Some(mt) = opt_match_token {
+            let mut ret = map();
+            let mut catch_all = false;
+
+            // FIXME: This should probably move _inside_ the Validator
+            // FIXME: Add precedences here
+            for mc in &mt.contents {
+                for item in &mc.items {
+                    // TODO: Mabye move this into MatchItem methods
+                    match *item {
+                        MatchItem::Unmapped(sym, _)        => { ret.insert(sym, sym); },
+                        MatchItem::Mapped(sym, mapping, _) => { ret.insert(mapping, sym); },
```

Comment:
I think we should initialize `all_literals` with the results from the `match_data` (using my new struct I proposed elsewhere) right here. Something like this:

```rust
// If we are not using an extern tokenizer, this map contains
// every literal (`"foo"`, `r"foo"`) that we may need to tokenize,
// mapping to a span where it appears in the source. It is filled
// in as we walk the `match` decl (if any) and the source itself.
let mut all_literals = map();
...
let (match_mappings, match_catch_all) = if let Some(mt) = opt_match_token {
    ...
    match *item {
        MatchItem::Unmapped(sym, span)        => {
            ret.insert(sym, sym);
            all_literals.insert(sym, span);
        }
        MatchItem::Mapped(sym, mapping, _) => {
            ret.insert(msppin, sym);
            all_literals.insert(sym, span);
        }
        MatchItem::CatchAll(_) => .. ,
    }
}
```